jetty-util/src/main/java/org/eclipse/jetty/util/DateCache.java

```diff
@@ -15,64 +15,80 @@
 
 import java.time.Instant;
 import java.time.ZoneId;
-import java.time.ZonedDateTime;
 import java.time.format.DateTimeFormatter;
 import java.util.Date;
 import java.util.Locale;
 import java.util.TimeZone;
 
 /**
- * Date Format Cache.
- * Computes String representations of Dates and caches
- * the results so that subsequent requests within the same second
- * will be fast.
- *
- * Only format strings that contain either "ss".  Sub second formatting is
- * not handled.
- *
- * The timezone of the date may be included as an ID with the "zzz"
- * format string or as an offset with the "ZZZ" format string.
- *
+ * Computes String representations of Dates then caches the results so
+ * that subsequent requests within the same second will be fast.
+ * <p>
  * If consecutive calls are frequently very different, then this
  * may be a little slower than a normal DateFormat.
+ * <p>
+ * @see DateTimeFormatter for date formatting patterns.
  */
 public class DateCache
 {
     public static final String DEFAULT_FORMAT = "EEE MMM dd HH:mm:ss zzz yyyy";
 
     private final String _formatString;
-    private final String _tzFormatString;
-    private final DateTimeFormatter _tzFormat;
-    private final Locale _locale;
+    private final DateTimeFormatter _tzFormat1;
+    private final DateTimeFormatter _tzFormat2;
     private final ZoneId _zoneId;
 
-    private volatile Tick _tick;
+    protected volatile Tick _tick1;
+    protected volatile Tick _tick2;
```

Comment:
BZZZZT BZZZT BZZZT ALERT ALERT ALERT!!!

One volatile is something to be noticed.  Two volatile variables is almost certainly a race.

Store both ticks in a record (or simple class) so that there is only a single volatile to check.